hazelcast/src/main/java/com/hazelcast/client/impl/protocol/ClientMessage.java

```diff
@@ -335,6 +334,26 @@ public ClientMessage copyWithNewCorrelationId(long correlationId) {
         return newMessage;
     }
 
+    /**
+     * Only deep copies the initial frame to not duplicated the rest of the
+     * message to get rid of unnecessary allocations for the retry of the same
+     * client message.
+     * <p>
+     * It is expected that the correlation id for the returned message is set
+     * later.
+     *
+     * @return the copied message
+     */
+    public ClientMessage copyWithUnsetCorrelationId() {
+        Frame initialFrameCopy = startFrame.deepCopy();
```

Comment:
this copies all the bytes including the old correlation id. hence, the method name is misleading. Your are actually doing a deep copy of the exact same message. you may change the method name? or unset the correlation id?

Reply:
Changed the name to `copyMessageWithSharedNonInitialFrames`. Is this OK?